23&22/scan.py

```diff
@@ -36,7 +36,7 @@ def retrieve(what):
     if (td != None):
         return td.get('content')
     else:
-        return "?"
+        return 0
```

Comment:
line 37 returns string, so here you should return '0' instead of integer.  For nutrition, 0 has meaning.  "Fat is 0" means really healthy, for example.  You probably want to default to something like '-99999'